lightning/src/routing/router.rs

```diff
@@ -917,7 +917,7 @@ where L::Target: Logger {
 					// Since we're choosing amount_to_transfer_over_msat as maximum possible, it can
 					// be only reduced later (not increased), so this channel should just be skipped
 					// as not sufficient.
-					if !over_path_minimum_msat && doesnt_exceed_cltv_delta_limit {
+					if contributes_sufficient_value && !over_path_minimum_msat && doesnt_exceed_cltv_delta_limit {
```

Comment:
Taking a stab at a clearer commit message (sorry long): "During the first pass of path finding, we seek a single path with the exact payment amount, and only seek additional paths if (a) no single path can carry the entire balance of the payment or (b) we found a good path, but along the way we found candidate paths with the potential to result in a lower total fee. This commit fixes the behavior of (b) -- we were previously considering some paths to be candidates for a lower fee when in fact they never would have worked. This caused us to re-run Dijkstra's when it might not have been beneficial."

Reply:
Thanks, @valentinewallace! @TheBlueMatt I took this for the commit message. Let me know if this sounds good to you.